flaml/onlineml/autovw.py

```diff
@@ -0,0 +1,157 @@
+import numpy as np
+from typing import Optional
+import logging
+from flaml.tune import Trial
+from flaml.onlineml import OnlineTrialRunner
+from flaml.scheduler import ChaChaScheduler
+from flaml.searcher import ChampionFrontierSearcher
+from flaml.onlineml.trial import get_ns_feature_dim_from_vw_example
+logger = logging.getLogger(__name__)
+
+
+class AutoVW:
+    """The AutoML class
+
+    Methods:
+        predict(data_sample)
+        learn(data_sample)
+    """
+    WARMSTART_NUM = 100
+
+    def __init__(self,
+                 init_config: dict,
+                 search_space: dict,
+                 max_live_model_num: int,
+                 min_resource_lease: Optional[float] = 'auto',
+                 automl_runner_args: Optional[dict] = {},
+                 scheduler_args: Optional[dict] = {},
+                 model_select_policy: Optional[str] = 'threshold_loss_ucb',
+                 metric: Optional[str] = 'mae_clipped',
+                 random_seed: Optional[int] = None,
+                 model_selection_mode: Optional[str] = 'min',
+                 cb_coef: Optional[float] = None,
+                 ):
+        """Constructor
+
+        Args:
+            init_config: A dictionary of a partial or full initial config,
+                e.g. {'interactions': set(), 'learning_rate': 0.5}
+            search_space: A dictionary of the search space. This search space includes both
```

Comment:
how do we distinguish between the two search_spaces (fixed and tunable?) does it matter at this stage?